alibi_detect/utils/loading.py

```diff
@@ -0,0 +1,50 @@
+# type: ignore
+from alibi_detect.version import __version__
+from alibi_detect.utils.schemas import DETECTOR_CONFIGS, DETECTOR_CONFIGS_RESOLVED, __config_spec__
+import warnings
+
+
+def validate_config(cfg: dict, resolved: bool = False) -> dict:
+    """
+    Validates a detector config dict by passing the dict to the detector's pydantic model schema.
+
+    Parameters
+    ----------
+    cfg
+        The detector config dict.
+    resolved
+        Whether the config is resolved or not. For example, if resolved=True, `x_ref` is expected to be a
+        np.ndarray, wheras if resolved=False, `x_ref` is expected to be a str.
+
+    Returns
+    -------
+    The validated config dict, with missing fields set to their default values.
+    """
+    # Get detector name
+    if 'name' in cfg:
+        detector_name = cfg['name']
+    else:
+        raise ValueError('`name` missing from config.toml.')
+
+    # Validate detector specific config
+    if detector_name in DETECTOR_CONFIGS.keys():
+        if resolved:
+            cfg = DETECTOR_CONFIGS_RESOLVED[detector_name](**cfg).dict()  # type: ignore[attr-defined]
+        else:
+            cfg = DETECTOR_CONFIGS[detector_name](**cfg).dict()  # type: ignore[attr-defined]
+    else:
+        raise ValueError('Loading the specified detector from a config.toml is not yet supported.')
```

Comment:
I guess this error would also be triggered if `name` is invalid/misspelled.

Reply:
Have changed the error to this:
```
raise ValueError(f'Loading the {detector_name} detector from a config.toml is not yet supported.')
```

so at least typo's will be a little more obvious to the user...